pkg/utils/kube_validation.go

```diff
@@ -0,0 +1,104 @@
+package utils
```

Comment:
I was reading how the use of catchall package names is discouraged. How about `pkg/kubeutils` ?

Reply:
Yeah, `utils` is no good. Really this should be under `internal/kube...`. If it's very specifically just utilities or validations we can do `kubeutils` or anything that describes what the package is for.